src/redux/slices/account/index.ts

```diff
@@ -0,0 +1,164 @@
+import Big from 'big.js';
+import { set } from 'lodash';
+import Router from 'next/router';
+
+import { reset as resetSegment } from '@/utils/analytics';
+
+import handleAsyncThunkStatus from '../../reducerStatus/handleAsyncThunkStatus';
+import initialStatusState from '../../reducerStatus/initialState/initialStatusState';
+
+interface handleSignIn {
+  accountId: string;
+  loading: string;
+  pretendAccountId: string;
+  signedAccountId: string;
+  state: string;
+  storageBalance: {
+    available: any;
+  };
+}
+
+const LS_ACCOUNT_ID = 'near-social-vm:v01::accountId:';
+const SLICE_NAME = 'account';
+
+const initialThunksStatusState = {
+  status: {
+    requestSignInWithWallet: initialStatusState.status,
+    logOut: initialStatusState.status,
+    refreshAllowance: initialStatusState.status,
+  },
+};
+
+const initialState = {
+  accountVM: {},
+  accountId: '',
+  availableStorage: '0',
+  storageCostPerByte: '',
+  ...initialThunksStatusState,
+
+  // TODO: move to a separate slice that will be responsible for loaders
+  showContent: false,
+};
+
+export const checkLoggedIn = createAsyncThunk(`${SLICE_NAME}/checkLoggedIn`, (_, { dispatch }) => {
+  const optimisticAccountId = window.localStorage.getItem(LS_ACCOUNT_ID);
+
+  !!optimisticAccountId && dispatch(setAccountId({ accountId: optimisticAccountId }));
+  dispatch(setShowContent({ showContent: true }));
+});
+
+export const handleSignIn = createAsyncThunk(
+  `${SLICE_NAME}/handleSignIn`,
+  ({ account, storageCostPerByte }: { account: handleSignIn; storageCostPerByte: any }, { dispatch }) => {
+    dispatch(
+      setAccount({
+        accountId: account?.accountId,
+        loading: account?.loading,
+        pretendAccountId: account?.pretendAccountId,
+        signedAccountId: account?.signedAccountId,
+        state: account?.state,
+        storageBalance: account?.storageBalance,
+      }),
+    );
+    dispatch(setAccountId({ accountId: account.accountId }));
+
+    // TODO: check the implementation
+    storageCostPerByte && dispatch(setStorageCostPerByte({ storageCostPerByte: storageCostPerByte.toString() }));
+
+    // TODO: move to utils
+    const availableStorage =
+      account.storageBalance && storageCostPerByte
+        ? Big(account.storageBalance.available).div(storageCostPerByte).toString()
+        : '0';
+
+    dispatch(setAvailableStorage({ availableStorage }));
+  },
+);
+
+export const logOut = createAsyncThunk(`${SLICE_NAME}/logOut`, async ({ near }: { near: any }, { dispatch }) => {
+  if (!near) {
+    return;
+  }
+
+  const wallet = await (await near.selector).wallet();
+  wallet.signOut();
+  near.accountId = null;
+  dispatch(setInitialState());
+  resetSegment();
+
+  // TODO: check if it's necessary
+  localStorage.removeItem('accountId');
+});
+
+export const requestSignInWithWallet = createAsyncThunk(
+  `${SLICE_NAME}/requestSignInWithWallet`,
+  async ({ near }: { near: any }) => {
```

Comment:
I'd like us to avoid `any` types but for now we can skip it

Reply:
Sure thing, I will add proper types - just didn't have enough time so far.